### PR TITLE
[11.0] [FIX] Counterpart's operating unit is wrong and Journals filtered by OUs

### PR DESCRIPTION
**Pull request by bjeficent**
The counterpart calculated in Internal Transfers had a wrong OU. The Journals were not filtered by OU's

### CODE REVIEW

File: account_operating_unit/models/account_invoice.py [hunk 1]
```diff
@@ -18,6 +18,26 @@ class AccountInvoice(models.Model):
...
+            rf = res.filtered(
+                    lambda aj: aj.operating_unit_id == self.operating_unit_id)
+            if not rf:
+                self.journal_id = res[0]
+            else:
+                self.journal_id = rf[0]
+
+    @api.onchange('journal_id')
+    def _onchange_journal(self):
+        if self.journal_id:
```
Comment (JordiBForgeFlow) [inline account_operating_unit/models/account_invoice.py]: `if self.journal_id and self.journal_id.operating_unit_id and self.journal_id.operating_unit_id != self.operating_unit_id'

File: account_operating_unit/models/account_invoice.py [hunk 2]
```diff
@@ -18,6 +18,26 @@ class AccountInvoice(models.Model):
...
+            res = self.env['operating.unit'].search(
+                [('id', '=', self.journal_id.operating_unit_id.id)]
```
Comment (JordiBForgeFlow) [inline account_operating_unit/models/account_invoice.py]: add `limit=1`

File: account_operating_unit/models/account_invoice.py [hunk 3]
```diff
@@ -18,6 +18,26 @@ class AccountInvoice(models.Model):
                                         states={'draft': [('readonly',
                                                            False)]})
 
+    @api.onchange('operating_unit_id')
+    def _onchange_operating_unit(self):
+        if self.operating_unit_id:
```
Comment (JordiBForgeFlow) [inline account_operating_unit/models/account_invoice.py]: `if self.operating_unit_id and (not self.journal_id or self.journal_id.operation_unit_id != self.operating_unit_id`

File: account_operating_unit/models/account_invoice.py [hunk 4]
```diff
@@ -18,6 +18,26 @@ class AccountInvoice(models.Model):
...
+            res = self.env['account.journal'].search(
+                [('type', '=', self.journal_id.type)])
+            rf = res.filtered(
+                    lambda aj: aj.operating_unit_id == self.operating_unit_id)
```
Comment (JordiBForgeFlow) [inline account_operating_unit/models/account_invoice.py]: Please change to `
journal = self.env['account.journal'].search([('type', '=', self.journal_id.type), ('operating_unit_id', '=', self.operating_unit_id.id), limit=1]
self.journal_id = journal
`

**Review by JordiBForgeFlow — CHANGES_REQUESTED** [review]

File: account_operating_unit/models/account_invoice.py (hunk 4 above)
Comment (AaronHForgeFlow) [reply account_operating_unit/models/account_invoice.py]: But if there is no specific journal for the OU, we can take a journal that is not assigned to any OU.

Comment (bjeficent) [reply account_operating_unit/models/account_invoice.py]: As @aheficent said, that way we could not take a journal without an assigned OU.

**Comment by bjeficent** [comment]
Comments attended, thanks for reviewing.

**Comment by AaronHForgeFlow** [comment]
Needs rebasing

**Comment by bjeficent** [comment]
Rebase done

**Pull request closed without merging**